src/applications/disability-benefits/all-claims/tests/utils/clean-toxic-exposure-data.unit.spec.jsx

```diff
@@ -0,0 +1,1466 @@
+      const formData = {
+        toxicExposure: {
+          conditions: { cancer: true },
+          gulfWar1990: {
+            bahrain: true,
+            iraq: false,
+            kuwait: true,
+            saudi: false,
+          },
+          gulfWar1990Details: {
+            bahrain: { startDate: '1990-01-01', endDate: '1990-12-31' },
+            iraq: { startDate: '1991-01-01', endDate: '1991-03-01' },
+            kuwait: { startDate: '1991-02-01', endDate: '1991-04-01' },
+            saudi: { startDate: '1991-01-01', endDate: '1991-06-01' },
+          },
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      // Should only keep details for selected locations
+      expect(result.toxicExposure.gulfWar1990Details).to.have.property(
+        'bahrain',
+      );
+      expect(result.toxicExposure.gulfWar1990Details).to.have.property(
+        'kuwait',
+      );
+      expect(result.toxicExposure.gulfWar1990Details).to.not.have.property(
+        'iraq',
+      );
+      expect(result.toxicExposure.gulfWar1990Details).to.not.have.property(
+        'saudi',
+      );
+    });
+
+    it('should completely remove toxicExposure when all data is empty', () => {
+      const formData = {
+        veteranInformation: { name: 'John Doe' },
+        toxicExposure: {
+          conditions: {},
+          gulfWar1990: {},
+          herbicide: null,
+          additionalExposures: undefined,
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      expect(result).to.not.have.property('toxicExposure');
+      expect(result.veteranInformation).to.deep.equal(
+        formData.veteranInformation,
+      );
+    });
+
+    it('should handle string and object formats for otherHerbicideLocations', () => {
+      // Test with string format (if supported)
+      const formDataString = {
+        toxicExposure: {
+          conditions: { asthma: true },
+          otherHerbicideLocations: 'Thailand base camps',
+        },
+      };
+
+      const resultString = cleanToxicExposureData(formDataString);
+      expect(resultString.toxicExposure.otherHerbicideLocations).to.equal(
+        'Thailand base camps',
+      );
+
+      // Test with empty string
+      const formDataEmptyString = {
+        toxicExposure: {
+          conditions: { asthma: true },
+          otherHerbicideLocations: '   ',
+        },
+      };
+
+      const resultEmptyString = cleanToxicExposureData(formDataEmptyString);
+      expect(resultEmptyString.toxicExposure).to.not.have.property(
+        'otherHerbicideLocations',
+      );
+    });
+
+    it('should handle empty conditions object', () => {
+      const formData = {
+        toxicExposure: {
+          conditions: {},
+          gulfWar1990: { bahrain: true },
+        },
+      };
+
+      const result = cleanToxicExposureData(formData);
+
+      expect(result).to.deep.equal(formData);
```

Comment:
Wouldn't this be an undesired result as there wouldn't be a condition or details associated with the location?